ocamltest/environments.ml

```diff
@@ -152,6 +189,7 @@ let rec apply_modifier environment = function
     apply_modifiers environment (find_modifiers modifiers_name)
   | Add (variable, value) -> add variable value environment
   | Append (variable, value) -> append variable value environment
-  | Remove variable -> remove variable environment
+    (* XXX Is this used any where - does it actually work? *)
+  | Remove variable -> VariableMap.remove variable environment
```

Comment:
Shouldn't this be updated to remove the variable from a pre-existing environment?
This is also a question for @shindere.

Reply:
I went through this in more detail - this facility is used [`recvfrom_linux.ml`](https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/lib-unix/unix-socket/recvfrom_linux.ml) here:

https://github.com/ocaml/ocaml/blob/ddd9ec2a91ebfbd5bb18e75d2d37efac5e3d3bfd/testsuite/tests/lib-unix/unix-socket/is-linux.sh#L12-L15

It is important that this actually _removes_ the binding completely (as opposed to _adding_ an instruction to `unsetenv` a variable) or this code fails:

https://github.com/ocaml/ocaml/blob/ddd9ec2a91ebfbd5bb18e75d2d37efac5e3d3bfd/ocamltest/actions_helpers.ml#L221-L230

My conclusion is that what I've done is correct, but changing `Environments.remove` was the wrong way to expose it - in 39d29be I have left `Environments.remove` alone and instead added `Environments.unsetenv`.

So the `Remove` action affects the _ocamltest_ environment (i.e. it removes a binding from the alterations ocamltest expects to make to the system environment when running a command). The new `unset` instruction _adds_ an instruction to the _ocamltest_ environment whose effect is to call `unsetenv` on any variable in the _system_ environment.